openstack/sharedfilesystems/v2/shares/results.go

```diff
@@ -110,3 +110,36 @@ type DeleteResult struct {
 type GetResult struct {
 	commonResult
 }
+
+// ExportLocation contains all information associated with a share export location
+type ExportLocation struct {
+	// The export location path that should be used for mount operation.
+	Path string `json:"path"`
+	// The UUID of the share instance that this export location belongs to.
+	ShareInstanceID string `json:"share_instance_id"`
+	// Defines purpose of an export location.
+	// If set to true, then it is expected to be used for service needs
+	// and by administrators only.
+	// If it is set to false, then this export location can be used by end users.
+	IsAdminOnly bool `json:"is_admin_only"`
+	// The share export location UUID.
+	ID string `json:"id"`
+	// Drivers may use this field to identify which export locations are
+	// most efficient and should be used preferentially by clients.
+	// By default it is set to false value. New in version 2.14
+	Preferred bool `json:"preferred"`
+}
+
+// ExtractExportLocations will get the Export Locations from the commonResult
+func (r commonResult) ExtractExportLocations() ([]ExportLocation, error) {
```

Comment:
Why is this a `commonResult` instead of a `GetExportLocationResult`? Can every response type extract this data?